src/device-api/index.ts

```diff
@@ -10,7 +10,7 @@ import * as apiKeys from '../lib/api-keys';
 import * as middleware from './middleware';
 
 import type { Server } from 'http';
-import type { AuthorizedRequest } from './types';
+import type { AuthorizedRequest } from '../types';
```

Comment:
I'm not sure about this commit. I think, whenever possible, we need to think as separate modules/subfolders as independent entities. Each module can have its own type definitions, its own internal helper functions, and their main API methods. 

Imho, a couple of good questions to ask are
- can I easily extract this submodule into its own repo? 
- does this piece of functionality of this submodule need to be exported? Is it used somewhere else in the code?

In this case, having the `AuthorizedRequest` as part of the main types does not help the rest of the code, so I don't think it belongs in `src/types`.

Reply:
That makes sense and was an oversight on my part, I'll move it to device-api.